File: Python Solutions/10.py
'''
Find the sum of all the primes below two million.
'''

def isPrime(n):
    if n == 2:
        return True

    for i in range(2, round(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

sum = 2

for i in range(3, 2000000, 2):
    if isPrime(i):
        sum += i;

print(sum)

File: C and C++ Solutions/10.c
/*
Find the sum of all the primes below two million.
*/
#include <stdio.h>
#include <math.h>

int isPrime(long long int n)
{
    if (n == 2)
    {
        return 1;
    }

    for (long long int i = 2; i < (long long int)pow(n, 0.5) + 2; i++)
    {
        if (n % i == 0)
        {
            return 0;
        }
    }
    return 1;
}

int main()
{
    long long int sum = 2;
    for (long long int i = 3; i < 2000000; i += 2)
    {
        if (isPrime(i))
        {
            sum += i;
        }
    }
    printf("%lld\n", sum);
}
